src/wallet/wallet.cpp

```diff
@@ -6663,6 +6663,14 @@ bool CWallet::InitLoadWallet(const CChainParams& params, bool clearWitnessCaches
 
     RegisterValidationInterface(walletInstance);
 
+    // Check for Orchard note commitment tree corruption and trigger a rescan
+    // if necessary.
+    auto orchardComplete = walletInstance->orchardWallet.CommitmentTreeComplete();
+    std::cout << "orchard commitment tree complete: " << orchardComplete << std::endl;
+    if (!orchardComplete) {
+        SoftSetBoolArg("-rescan", true);
```

Comment:
It would be a good idea to print in the log file that we're forced to perform a rescan and for what reason, so people aren't confused about why it's rescanning unprompted.

Reply:
To avoid an extra CI roundtrip, let's add this logging as part of creating the release.